.github/workflows/build-test.yml

```diff
@@ -0,0 +1,53 @@
+name: Check
+
+on:
+  pull_request:
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-code-style:
+    name: Code Style
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt scalafmtCheckAll scalafmtSbtCheck
+
+#  MiMa is not enabled yet, since this library is marked as "API may change".
+#  check-binary-compatibility:
+#    name: Binary Compatibility
+#    uses: playframework/.github/.github/workflows/binary-check.yml@v1
+
+  check-docs:
+    name: Docs
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt doc
+
+  check-headers:
+    name: Headers
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt headerCheck test:headerCheck
```

Comment:
Instead of `headerCheck test:headerCheck` you can just use `headerCheckAll` in newer sbt-header versions now.